ncompass_rust/trace_converters/nsys-chrome/tests/test_mapping.rs

```
@@ -0,0 +1,411 @@
+//! Unit tests for mapping module
+
+use nsys_chrome::mapping::{decompose_global_tid, extract_device_mapping, extract_thread_names, get_all_devices};
+use rusqlite::Connection;
+use tempfile::NamedTempFile;
+
+// ==========================
+// Tests for decompose_global_tid
+// ==========================
+
+#[test]
+fn test_decompose_global_tid_basic() {
+    // globalTid = (PID << 24) | TID
+    // PID = 1, TID = 100 -> globalTid = (1 << 24) | 100 = 16777316
+    let global_tid: i64 = (1 << 24) | 100;
+    let (pid, tid) = decompose_global_tid(global_tid);
+    assert_eq!(pid, 1);
+    assert_eq!(tid, 100);
+}
+
+#[test]
+fn test_decompose_global_tid_zero() {
+    let (pid, tid) = decompose_global_tid(0);
+    assert_eq!(pid, 0);
+    assert_eq!(tid, 0);
+}
+
+#[test]
+fn test_decompose_global_tid_max_tid() {
+    // Max TID is 0xFFFFFF (24 bits)
+    let global_tid: i64 = (1 << 24) | 0xFFFFFF;
+    let (pid, tid) = decompose_global_tid(global_tid);
+    assert_eq!(pid, 1);
+    assert_eq!(tid, 0xFFFFFF);
+}
+
+#[test]
+fn test_decompose_global_tid_max_pid() {
+    // Max PID is also 0xFFFFFF (24 bits extracted)
+    let global_tid: i64 = (0xFFFFFF_i64 << 24) | 100;
+    let (pid, tid) = decompose_global_tid(global_tid);
+    assert_eq!(pid, 0xFFFFFF);
+    assert_eq!(tid, 100);
+}
+
+#[test]
+fn test_decompose_global_tid_large_value() {
+    // Test with a realistic large value
+    let global_tid: i64 = 12345678901234;
+    let (pid, tid) = decompose_global_tid(global_tid);
+    
+    // Verify the decomposition is reversible
+    let reconstructed = ((pid as i64) << 24) | (tid as i64);
+    // Due to masking, we only get lower 48 bits
+    let expected = global_tid & 0xFFFFFFFFFFFF;
+    assert_eq!(reconstructed, expected);
```

[QUESTION] same silent fail question

---

intentional